src/main.cpp

```diff
@@ -4605,6 +4630,13 @@ bool ReceivedBlockTransactions(
         // valueBalanceOrchard behaves the same way as valueBalanceSapling.
         orchardValue += -tx.GetOrchardBundle().GetValueBalance();
 
+        // Each funding stream produces a positive change to the lockbox value.
+        for (auto elem : chainparams.GetConsensus().GetActiveFundingStreamElements(pindexNew->nHeight)) {
+            if (std::holds_alternative<Consensus::Lockbox>(elem.first)) {
+                lockboxValue += elem.second;
+            }
+        }
```

Comment:
We only need to do this either here (`ReceivedBlockTransactions`) or above (in `ConnectBlock`); we should not need to do both, as these are both called in the "received block from network -> connected to chain" online pathway.

Note that here we (re)set `nChainTotalSupply` to `nullopt` (to ensure it isn't accidentally set), but don't set either `nTotalSupply` or `nChainTotalSupply` until `ConnectBlock` (because we need context in order to compute the transparent delta).

Conversely, here we set `nOrchardValue`, and we _also_ set `nChainOrchardValue` further down, because we can do that non-contextually.

Currently, this is the only place we use `pindexNew->nHeight` directly in `ReceivedBlockTransactions` (we process potentially-linked descendents below, but we do not use anything height-dependent). We cannot set `lockboxValue` truly non-contextually (we need height information), but we _can_ set it without referencing height-specific state (as is necessary for transparent), so it's maybe okay to set everything in `ReceivedBlockTransactions`? The advantage of that is we have the lockbox data set for all blocks in which we receive the full block, whereas if we set in `ConnectBlock` we only have the lockbox data set for blocks that have been connected to a main chain at least once.